cohort-evaluator-spark/src/test/java/com/ibm/cohort/cql/spark/BaseSparkTest.java

```diff
@@ -4,38 +4,50 @@
 import java.util.Map;
 
 import org.apache.spark.sql.SparkSession;
+import org.apache.spark.sql.delta.DeltaLog;
+import org.junit.Before;
 
-public class BaseSparkTest implements Serializable {
+/**
+ * A base class that provides common functions for unit testing with Spark.
+ *
+ * All test classes using a {@link SparkSession} should extend and use this class.
+ *
+ * Unless required, you should _not_ close a {@link SparkSession} provided by this class.
+ * We're trying to leverage Spark's session caching to limit how much time is spent
+ * spinning up sessions throughout our test suite.
+ */
+public abstract class BaseSparkTest implements Serializable {
     private static final long serialVersionUID = 1L;
 
-    public static enum Java8API {
+    public enum Java8API {
```

Comment:
Is there a specific reason you made this instance-level and opened up the constructor?

Reply:
IntelliJ was giving me warnings since enums have their own rules with what scope they exist in and the accessibility of their constructors.  I can add them back if you would prefer to be more explicit.